paasta_tools/tron_tools.py

```diff
@@ -1180,15 +1180,57 @@ def validate_complete_config(
     return []
 
 
-def get_tron_namespaces(cluster, soa_dir):
+def _is_valid_namespace(
+    job: {str: TronActionConfigDict}, tron_executors: List[str]
+) -> bool:
+    for _, action_info in job.get("actions", {}).items():
+        if "paasta" in tron_executors and action_info.get("executor", "") in [
+            "paasta",
+            "",
+        ]:
+            return True
+        elif (
+            "spark" in tron_executors
+            and action_info.get("executor", "") == "ssh"
+            and "spark-run" in action_info.get("command", "")
+        ):
+            return True
+        elif (
+            "ssh_not_spark" in tron_executors
+            and action_info.get("executor", "") == "ssh"
+            and "spark-run" not in action_info.get("command", "")
+        ):
+            return True
+    return False
+
+
+def get_tron_namespaces(
+    cluster: str,
+    soa_dir: str,
+    tron_executors: List[str] = ["paasta"],
+) -> List[str]:
     tron_config_file = f"tron-{cluster}.yaml"
     config_dirs = [
         _dir[0]
         for _dir in os.walk(os.path.abspath(soa_dir))
         if tron_config_file in _dir[2]
     ]
     namespaces = [os.path.split(config_dir)[1] for config_dir in config_dirs]
-    return namespaces
+    tron_namespaces = set()
+    for namespace in namespaces:
+        config = filter_templates_from_config(
+            read_extra_service_information(
+                namespace,
+                extra_info=f"tron-{cluster}",
+                soa_dir=soa_dir,
+                deepcopy=False,
+            )
+        )
+        for _, job in config.items():
+            if _is_valid_namespace(job, tron_executors):
```

Comment:
i think that once we just go towards what we talked about in slack, this can just be something like 
```python
...
all_namespaces = [os.path.split(config_dir)[1] for config_dir in config_dirs]
filtered_namespaces = set()

for job in load_tron_service_config(...):
    for action in job.get_actions():
        if action.get_executor() in executor_types:
            filtered_namespaces.add(job.service)

return list(filtered_namespaces)
```